src/Desktop.Analyzers/UnitTests/DoNotUseInsecureDTDProcessingUseXmlReaderForXPathDocumentTests.cs

```diff
@@ -0,0 +1,366 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.UnitTests;
+using Microsoft.CodeAnalysis.CSharp;
+using Xunit;
+
+namespace Desktop.Analyzers.UnitTests
+{
+    public partial class DoNotUseInsecureDTDProcessingAnalyzerTests : DiagnosticAnalyzerTestBase
+    {
+        private DiagnosticResult GetCA3075XPathDocumentCSharpResultAt(int line, int column, string name)
+        {
+            return GetCSharpResultAt(line, column, CA3075RuleId, string.Format(CA3075LoadXmlMessage, name, ".ctor"));
+        }
+
+        private DiagnosticResult GetCA3075XPathDocumentBasicResultAt(int line, int column, string name)
+        {
+            return GetBasicResultAt(line, column, CA3075RuleId, string.Format(CA3075LoadXmlMessage, name, ".ctor"));
+        }
+
+        [Fact]
+        public void UseXPathDocumentShouldGenerateDiagnostic()
```

Comment:
It's not the use of `XPathDocument` _per se_ that generates the diagnostic, it's the fact that you don't specify a reader that prohibits DTD processing. So the test name might be `XPathDocumentWithoutSecureReaderShouldGenerateDiagnostic`.